### PR TITLE
[MODULAR] Security Panzerfaust.

### PR DESCRIPTION
**Pull request by SpaceVampire**
## About The Pull Request
Does EXACTLY as the title says, makes a new rocket launcher subtype with a new look and mindshield firing pin, and then makes a whole roster of new rockets for them to fire at criminals (definitely criminals). Adds them to the station armory, with the printing of additional ammo locked behind advanced weapon tech.

Smoke Rocket:
Does exactly what you think it would, makes a big cloud of smoke and hurts to get hit directly with.
White Phosphorus Rocket:
A similar effect to the smoke rocket, however it creates some fire and has less smoke range.
Sleeping Smoke Rocket:
For some reason we have a smoke subtype that just makes you sleep, so do with that what you will.
EMP Rocket:
This has replaced the ion rifle in every armory, as now we have missiles that make big emp bursts on launch.
APCBCFSDSGLATM:
The "Fuck you" missile, WILL delimb whoever it hits, bounces fucking everywhere, only to be used for dire circumstances.

(Port of https://github.com/tgstation/tgstation/pull/61234, because some ideas are just too good to be left to waste).
## How This Contributes To The Skyrat Roleplay Experience

Given the recent complaints of security being understaffed, and under-powered, I think that this shift into a new weapon paradigm will allow for additional roleplay episodes and HRP moments. Giving some new life to the department. It would certainly motivate me to play it more, because rockets are really cool. I went ahead and tested it, and everything seems fine and balanced^tm! I think at the very least this is worthy of the high honor of a test-merge, to lighten up all of our spirits in these trying times. To remember the hard work of our men in blue, and all they do for the station!\

EDIT: As per HeadCoder request, it is now done in classic Panzerfaust style.

## Changelog
:cl:
add: Five new flavors of 84mm rocket
add: A security made rocket launcher to use these (sprites from tgmc)
qol: People will learn to respect the authority!
fix: Every single sec vs antag balance issue is now fixed
/:cl:

### CODE REVIEW

**Comment by Paxilmaniac** [comment]
sigma rocket launcher porter

File: modular_skyrat/modules/rocket-launcher/rocket-launcher.dm [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+		/obj/item/ammo_casing/caseless/rocket/sabot = 2,
+		/obj/item/ammo_casing/caseless/rocket/smoke/phosphor = 1,
+		/obj/item/ammo_casing/caseless/rocket/smoke/sleeping = 1,
+		/obj/item/ammo_casing/caseless/rocket/emp = 1,
+	), src)
+
+/obj/item/ammo_casing/caseless/rocket/sabot
+	name = "\improper APCBCFSDSGLATM"
+	desc = "Why anyone would name a rocket 'Armour Piercing Capped Ballistic Capped Fin Stabilized Discarding Sabot Gun Launched Anti Tank Missile' is beyond your understanding."
+	icon = 'modular_skyrat/modules/rocket-launcher/icon/rocketsec-ammo.dmi'
```
Comment (FlamingLily) [inline modular_skyrat/modules/rocket-launcher/rocket-launcher.dm:30]:
```modular_skyrat/modules/rocket-launcher/rocket-launcher.dm:71:error: 'modular_skyrat/modules/rocket-launcher/icon/rocketlaunchersec.dmi': cannot find file

modular_skyrat/modules/rocket-launcher/rocket-launcher.dm:72:error: 'modular_skyrat/modules/rocket-launcher/icon/rocketsec-right.dmi': cannot find file

modular_skyrat/modules/rocket-launcher/rocket-launcher.dm:73:error: 'modular_skyrat/modules/rocket-launcher/icon/rocketsec-left.dmi': cannot find file

modular_skyrat/modules/rocket-launcher/rocket-launcher.dm:30:error: 'modular_skyrat/modules/rocket-launcher/icon/rocketsec-ammo.dmi': cannot find file

modular_skyrat/modules/rocket-launcher/rocket-launcher.dm:38:error: 'modular_skyrat/modules/rocket-launcher/icon/rocketsec-ammo.dmi': cannot find file

modular_skyrat/modules/rocket-launcher/rocket-launcher.dm:46:error: 'modular_skyrat/modules/rocket-launcher/icon/rocketsec-ammo.dmi': cannot find file

modular_skyrat/modules/rocket-launcher/rocket-launcher.dm:54:error: 'modular_skyrat/modules/rocket-launcher/icon/rocketsec-ammo.dmi': cannot find file

modular_skyrat/modules/rocket-launcher/rocket-launcher.dm:61:error: 'modular_skyrat/modules/rocket-launcher/icon/rocketsec-ammo.dmi': cannot find file

modular_skyrat/modules/rocket-launcher/rocket-launcher.dm:83:error: 'modular_skyrat/modules/rocket-launcher/icon/rocketsec-project.dmi': cannot find file

modular_skyrat/modules/rocket-launcher/rocket-launcher.dm:101:error: 'modular_skyrat/modules/rocket-launcher/icon/rocketsec-project.dmi': cannot find file

modular_skyrat/modules/rocket-launcher/rocket-launcher.dm:114:error: 'modular_skyrat/modules/rocket-launcher/icon/rocketsec-project.dmi': cannot find file

modular_skyrat/modules/rocket-launcher/rocket-launcher.dm:131:error: 'modular_skyrat/modules/rocket-launcher/icon/rocketsec-project.dmi': cannot find file

modular_skyrat/modules/rocket-launcher/rocket-launcher.dm:145:error: 'modular_skyrat/modules/rocket-launcher/icon/rocketsec-project.dmi': cannot find file```

Comment (SpaceVampire) [reply modular_skyrat/modules/rocket-launcher/rocket-launcher.dm:30]: Yeah, I know, I know.
I didn't notice that my Github desktop didn't mark em correctly.
It all complied and worked fine when I tested it locally.
Just need to push the right stuff here.

File: _maps/map_files/IceBoxStation/IceBoxStation_skyrat.dmm [hunk 2]
```diff
@@ -54483,6 +54483,16 @@
 /obj/effect/spawner/structure/window,
 /turf/open/floor/plating,
 /area/service/hydroponics/garden)
+"vkc" = (
+/obj/item/storage/belt/military/rocket/full,
+/obj/item/storage/belt/military/rocket/full,
+/obj/item/storage/belt/military/rocket/full,
+/obj/item/gun/ballistic/rocketlauncher/secuirty,
+/obj/item/gun/ballistic/rocketlauncher/secuirty,
```
Comment (Vizoplez) [inline _maps/map_files/IceBoxStation/IceBoxStation_skyrat.dmm:54491]: Bruh. Security misspelled

File: modular_skyrat/modules/rocket-launcher/rocket-launcher.dm [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+	name = "\improper GLEMPS"
+	desc = "This rocket seems to be packed full of machinery and a special chemical mix that makes a large emp on impact."
+	icon = 'modular_skyrat/modules/rocket-launcher/icon/rocketsec-ammo.dmi'
+	icon_state = "84mm-emp"
+	projectile_type = /obj/projectile/bullet/a84mm/emp
+	w_class = WEIGHT_CLASS_BULKY
+
+/obj/item/ammo_casing/caseless/rocket/smoke
+	name = "\improper HVSD"
+	desc = "The acronym stands for high velocty smoke deployment, surely that makes it non-lethal, right?"
```
Comment (tf-4) [inline modular_skyrat/modules/rocket-launcher/rocket-launcher.dm]:
```suggestion
	desc = "The acronym stands for 'high velocity smoke deployment', surely that makes it non-lethal, right?"
```

File: _maps/map_files/NSSJourney/NSSJourney.dmm [hunk 4]
```diff
@@ -53821,6 +53821,16 @@
 	},
 /turf/open/floor/iron,
 /area/engineering/atmos)
+"udf" = (
+/obj/structure/rack,
+/obj/item/gun/ballistic/rocketlauncher/secuirty,
+/obj/item/gun/ballistic/rocketlauncher/secuirty,
+/obj/item/gun/ballistic/rocketlauncher/secuirty,
+/obj/item/storage/belt/military/rocket/full,
+/obj/item/storage/belt/military/rocket/full,
```
Comment (Vizoplez) [inline _maps/map_files/NSSJourney/NSSJourney.dmm:53830]: Don't give them three launchers and 3 belts ffs. Give them 1

File: modular_skyrat/modules/rocket-launcher/rocket-launcher.dm [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+	icon = 'modular_skyrat/modules/rocket-launcher/icon/rocketsec-ammo.dmi'
+	icon_state = "84mm-sleepsmoke"
+	projectile_type = /obj/projectile/bullet/a84mm/smoke/sleeping
+
+/obj/item/ammo_box/magazine/internal/rocketlauncher/smoke
+	ammo_type = /obj/item/ammo_casing/caseless/rocket/smoke
+
+/obj/item/gun/ballistic/rocketlauncher/secuirty
+	name = "\improper UPL-550"
+	desc = "A nanotrasen made 'utility launcher' which in reality, is just a fancy way to say legalized rocket launcher."
```
Comment (tf-4) [inline modular_skyrat/modules/rocket-launcher/rocket-launcher.dm]:
```suggestion
	desc = "A nanotrasen made 'utility launcher' which, in reality, is just a fancy way to say legalized rocket launcher."
```

File: modular_skyrat/modules/rocket-launcher/rocket-launcher.dm [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+	lefthand_file = 'modular_skyrat/modules/rocket-launcher/icon/rocketsec-left.dmi'
+	icon_state = "rocketlauncher-sec"
+	inhand_icon_state = "rocketlauncher-sec"
+	pin = /obj/item/firing_pin/implant/mindshield
+	mag_type = /obj/item/ammo_box/magazine/internal/rocketlauncher/smoke
+
+/// Tacticool (Useless) ammo for the PML and subtypes
+/obj/projectile/bullet/a84mm/sabot
+	name = "\improper APCBCFSDSGLAT missile"
+	desc = "Pray this isn't coming straigh at you."
```
Comment (tf-4) [inline modular_skyrat/modules/rocket-launcher/rocket-launcher.dm]:
```suggestion
	desc = "Pray this isn't coming straight at you."
```

**Review by Vizoplez — CHANGES_REQUESTED** [review]

**Comment by Vizoplez** [comment]
Other than that good 2 merge imo

**Review by tf-4 — COMMENTED** [review]
👍

File: _maps/map_files/IceBoxStation/IceBoxStation_skyrat.dmm (hunk 2 above)
Comment (GoldenAlpharex) [reply _maps/map_files/IceBoxStation/IceBoxStation_skyrat.dmm:54491]: *Secuirty, get it right Yawet

File: _maps/map_files/NSSJourney/NSSJourney.dmm (hunk 4 above)
Comment (SpaceVampire) [reply _maps/map_files/NSSJourney/NSSJourney.dmm:53830]: Given that you can only print more ammo for it, and not more weapons.
I might go ahead and just knock it down to two starting instead.
Would make it one, but then I would want to add em to cargo.

File: _maps/map_files/IceBoxStation/IceBoxStation_skyrat.dmm (hunk 2 above)
Comment (SpaceVampire) [reply _maps/map_files/IceBoxStation/IceBoxStation_skyrat.dmm:54491]: That is what I get for no double-checking at 3:00am, lamo.

**Comment by Lauwma** [comment]
A new tool to fight the horny in brig.
I'm concerned with the phosphorus rocket though, I hope it's not a big fire because it could definitely lead to some "HRP moments"

**Comment by Paxilmaniac** [comment]
> A new tool to fight the horny in brig.
> I'm concerned with the phosphorus rocket though, I hope it's not a big fire because it could definitely lead to some "HRP moments"

The phosphorus rocket is nothing entirely special, or especially dangerous, mainly because I was too lazy to make it so when coding it. The fire part is literally just an explosion with no explosive range, and fire range that fills the inside of the smoke.
Basically just means that the fire isn't plasma or anything, so it should be fine.

**Comment by KathrinBailey** [comment]
I can't tell if you're being serious or not

**Comment by SpaceVampire** [comment]
> I can't tell if you're being serious or not

I'm absolutely serious, with only the utmost sincerity.
After reading over my "How This Contributes To The Skyrat Roleplay Experience" once again.
I have found that it resonates with my very soul, even more than before.

**Comment by ForrestWick** [comment]
Security has a grenade launcher

**Comment by SpaceVampire** [comment]
I'm probably going to update the sprites/themes for it more.
Because they still don't match exactly what I now have in mind.
If I don't end up closing this first due to either maintainer disagreement, lack of feedback, or loss of interest.

**Comment by KathrinBailey** [comment]
I mean Forrest kinda beat me to it.
Security *has* a grenade launcher that can fire 40mms and chem nades, not sure why security need a new platform to fire these shells that I'm not even sure they should have. EMP especially. I think the ol' willy pete might be a bit too much of a meme, not sure how you'd get away having that... _or why you'd ever need it?_
I'm just not seeing why we need it when we could just add new grenades in?

**Comment by SpaceVampire** [comment]
People seem to enjoy the general idea of this, but uh.
Given that Maintainers don't seem to like the current state of it.
I'm going to close this for the moment until I have enough time to come up with a better implementation.
Probably something locked behind Cargo, a new kind of token, or like.
Reworked into an antag thing, instead of a security thing.
Wanna focus more on other stuff right now.

File: modular_skyrat/modules/rocket-launcher/rocket-launcher.dm [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+	righthand_file = 'modular_skyrat/modules/rocket-launcher/icon/rocketsec-right.dmi'
+	lefthand_file = 'modular_skyrat/modules/rocket-launcher/icon/rocketsec-left.dmi'
+	icon_state = "rocketlauncher-sec"
+	inhand_icon_state = "rocketlauncher-sec"
+	pin = /obj/item/firing_pin/implant/mindshield
+	mag_type = /obj/item/ammo_box/magazine/internal/rocketlauncher/smoke
+
+/// Tacticool (Useless) ammo for the PML and subtypes
+/obj/projectile/bullet/a84mm/sabot
+	name = "\improper Panzerfaust-BLUNT"
```
Comment (Iamgoofball) [inline modular_skyrat/modules/rocket-launcher/rocket-launcher.dm:81]: Get this Wehraboo shit out of here.

**Comment by Lauwma** [comment]
Stop locking things behind cargo because it makes so tools are never used. You know why ? Because ordering things to stuff the armory (without using them) is considered as powergaming. When you order your tools it's already too late either the threat prevailed either the threat already been neutralized. Tools are useless if You can't have them when you need them, thus why they should start in the armory.
(It's not like it's hard to break into the armory and steal it as antag)

**Pull request closed without merging**